Sources/xcproj/PBXProj.swift

```diff
@@ -22,20 +22,20 @@ public class PBXProj: Decodable {
     public var aggregateTargets: [PBXAggregateTarget] = []
     public var containerItemProxies: [PBXContainerItemProxy] = []
     public var copyFilesBuildPhases: [PBXCopyFilesBuildPhase] = []
-    public var groups: [PBXGroup] = []
+    public var groups: [String: PBXGroup] = [:]
```

Comment:
This will be a breaking change. We'll have to do a major version bump. What do you think @yonaskolb ?

Reply:
Yeah, it will be a breaking change. There is the option of creating these dictionary caches privately and on demand when we need them, which is at the beginning of writing. But perhaps have them as dictionaries is a better public API as well? 
Another side effect of this is that all duplicate references get removed automatically. This is mostly good, but means tools like `xclint` can't show you which references are duplicated. Is that ok?

In either case, I think all the properties should match, and if the public api is changing they should then all be changed to dictionaries (including getters like `buildPhases` and `objects`).

Reply:
What does this specifically break? I tested with XCodeGen so I'm assuming it's other projects in the xcodeswift org?

Reply:
It breaks any project that reads or writes these properties and expects them to be arrays.
eg:
```
pbxProj.fileReferences.append(myFile)
```
XcodeGen is ok as it uses `func addObject(_ object: PBXObject)`